scene/3d/cpu_particles.cpp

```diff
@@ -932,7 +928,7 @@ void CPUParticles::_update_particle_data_buffer() {
 					Vector3 dir = c->get_global_transform().basis.get_axis(2); //far away to close
 
 					if (local_coords) {
-						dir = un_transform.basis.xform(dir).normalized();
+						dir = Basis().xform(dir).normalized();
```

Comment:
```suggestion
						dir = dir.normalized();
```
Wouldn't that get us the same result? Also, I find it weird that the direction gets normalized only in local coords but not in global coords. We may be able to skip this normalization, or maybe we have to normalize in both cases, but the distinction seems supicious.

Reply:
I feel the same way. I reduced it to this so that I could get rid of ``un_transform``, but I wanted to leave it as intact as possible. 

I'll compare it to the code for particles and see if that exposes any reason for normalizing ``dir`` only when using ``local_coords``.

Reply:
I got it ``sorted`` out. :p 

Here is the implementation for Particles
https://github.com/godotengine/godot/blob/f410e7a8a9b14fd9f5c210afcc9e3519e9ce246d/drivers/gles3/rasterizer_scene_gles3.cpp#L1462-L1466

And here is how I updated it
```
if (local_coords) {

	// will look different from Particles in editor as this is based on the camera in the scenetree
	// and not the editor camera
	dir = get_global_transform().affine_inverse().xform(dir).normalized();
} else {
	dir = dir.normalized();
}
```
I also tested out and confirmed that the previous implementation did not match the behaviour of Particles. It is fixed now. Thanks for drawing attention to this.

Reply:
Great work!